apps/prairielearn/src/sync/course-db.ts

```diff
@@ -711,20 +708,34 @@ async function loadAndValidateJson<T extends { uuid: string }>({
     return loadedJson;
   }
 
-  loadedJson.data = { ...defaults, ...loadedJson.data };
+  // If we didn't get any errors with the ajv schema, we will re-parse with Zod, which will fill in default values and let us
+  // use the output type.
+  const result = zodSchema.safeParse(loadedJson.data);
+  if (!result.success) {
+    infofile.addErrors(
+      loadedJson,
+      result.error.issues.map(
+        (e) =>
+          `code: ${e.code}, path: ${e.path.join('.')}, message: ${e.message}. Report this error to the PL team, this should not happen.`,
```

Comment:
Nit, we never want to use "PL" in the UI (or in docs, really):

```suggestion
          `code: ${e.code}, path: ${e.path.join('.')}, message: ${e.message}. Report this error to the PrairieLearn team, this should not happen.`,
```

We might consider a `Sentry.captureException` here too? 🤷

Reply:
Added the Sentry.captureException in fea8fd1a66276bbfc030e19a9b078abbde6f3b52